debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -274,6 +274,12 @@ protected Optional<ErrorHandler> getErrorHandler() {
 
     @Override
     protected void doStop() {
+        // Shut down the queue early to prevent memory leaks from blocked threads
+        if (queue != null) {
+            LOGGER.info("Shutting down queue to prevent thread leaks");
+            queue.shutdown();
+        }
```

Comment:
Would this also be necessary for MariaDB? If so, it may be useful to create a `doStop()` implementation in the `BinlogSourceTask` class and place this logic there, having both MySQL and MariaDB call their respective super type methods.  What do you think?